packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -302,7 +307,11 @@ export async function runHttpQuery(
           context = await context();
         } catch (e) {
           e.message = `Context creation failed: ${e.message}`;
-          throwHttpGraphQLError(500, [e], optionsObject);
+          if (e.extensions && e.extensions.code) {
+            throwHttpGraphQLError(400, [e], optionsObject);
```

Comment:
What is the thinking here? Why are errors that set a `code` always 400?

Reply:
The thinking is that errors with codes are thrown by the user and are generally not internal server errors. I think adding a check for non `INTERNAL_SERVER_ERROR` makes sense

Reply:
Hmmm, I don't think that really holds though. 4xx errors are client errors, but not all errors thrown by user code are client errors. If a backend is down for example, that would be more appropriately modeled as a 5xx error (maybe similar to 503 service unavailable).

I don't think this is super relevant because GraphQL responses  don't map cleanly to status codes anyway, but 400 seems like an odd default because it signals `Bad Request`, and that puts the blame on the client. A client may want to retry 5xx requests for example, but not do this for a 400.